### PR TITLE
Various changes based on standardization

### PR DESCRIPTION
**Pull request by Nathan-Binkley**

### CODE REVIEW

File: bot/cogs/grades_data/grades_cog.py [hunk 1]
```diff
@@ -277,5 +341,42 @@ async def grades(self, ctx, course, year=MIN_YEAR):
 
         await ctx.send(embed=embed)
 
+    @commands.command()
+    async def prof(self, ctx, firstName, lastName, detailed = bool('yes')):
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:345]: d.py exposes a boolean type hint converter for command args. relevant docs: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#bool

File: bot/cogs/grades_data/grades_cog.py [hunk 2]
```diff
@@ -277,5 +341,42 @@ async def grades(self, ctx, course, year=MIN_YEAR):
...
+        """
+        #Handle casing
+        prof_name = f'{firstName.lower().capitalize()} {lastName.lower().capitalize()}' 
+        
+        if prof_name not in self.global_master_prof_list:
+            embed = discord.Embed(title="Grades", color=Colors.Error)
+            result = 'That\'s not a professor at Clemson\n Are you sure you used the proper notation (ex: Brian Dean)?'
+            embed.add_field(name="ERROR: Professor doesn't exist", value=result, inline=False)
+            await ctx.send(embed=embed)
+        else:
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:369]: to reduce nesting we can put a return after line 368 then remove this else, turn it from an if else into an early exit condition which is the standard pattern in clembot. not a big deal just cuts down on nesting optionally

File: bot/cogs/grades_data/grades_cog.py [hunk 3]
```diff
@@ -277,5 +341,42 @@ async def grades(self, ctx, course, year=MIN_YEAR):
...
+        EX: !prof Brian Dean                <-- will print all letter distributions
+        !prof Brian Dean blah blah blah     <-- will print only pass/fail/withdrawal
+
+        DISCLAIMER:
+        Not every professor listed will be at Clemson, this is a tool built for better information but not complete information
+        In addition, this system works on the Grade Distribution Releases located at https://www.clemson.edu/institutional-effectiveness/oir/data-reports/
+        *Course Sections that meet the following conditions are not included: Undergraduate classes with less than 10 students or Graduate classes with less than 5 students. In addition, if a section has all but 1 student making a common grade (example: All but one student makes a "B" in a class), the section is excluded.*
+        """
+        #Handle casing
+        prof_name = f'{firstName.lower().capitalize()} {lastName.lower().capitalize()}'
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:362]: this is clever 😁

Comment (Nathan-Binkley) [reply bot/cogs/grades_data/grades_cog.py:362]: I mean, it works lol

File: bot/cogs/grades_data/grades_cog.py (hunk 1 above)
Comment (Nathan-Binkley) [reply bot/cogs/grades_data/grades_cog.py:345]: So change something?

Comment (Jay-Madden) [reply bot/cogs/grades_data/grades_cog.py:345]: ya, make it a boolean converter. Thats much more flexible then a string

File: bot/cogs/grades_data/grades_cog.py [hunk 4]
```diff
@@ -234,6 +244,61 @@ def searchCourse(self, query, year):
...
+            courses[course] = {
+                "A": [],
+                "B": [],
+                "C": [],
+                "D": [],
+                "F": [],
+                "W": []
+            }
+
+        for section in primary:
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:263]:
```suggestion

```

File: bot/cogs/grades_data/grades_cog.py [hunk 5]
```diff
@@ -234,6 +244,61 @@ def searchCourse(self, query, year):
...
+            course = section['course'] + "-" + section['number']
```
Comment (Jay-Madden) [inline bot/cogs/grades_data/grades_cog.py:264]:
```suggestion

```

**Review by Jay-Madden — APPROVED** [review]

**Pull request merged**